JOSS paper preparation (#1249)
earthaccess-dev/earthaccess: paper/paper.md

Author: @mfisher87, want to create an issue for it?

Author: @mfisher87 – pinging you again on this :)

    orcid: "https://orcid.org/0000-0002-3039-0260"
    affiliation: "1"
  - given-names: "Sherwin"
    surname: "Varghese"
    orcid: "https://orcid.org/0009-0005-7030-213X"
    affiliation: "9"
affiliations:
 - name: "National Snow and Ice Data Center (NSIDC), Cooperative Institute for Research in Environmental Sciences (CIRES), University of Colorado Boulder, Boulder, CO, USA"
   index: 1
   ror: 02s376052
 - name: "Goddard Space Flight Center, National Aeronautics and Space Administration, Greenbelt, MD, USA"
   index: 2
   ror: 0171mag52
 - name: "Coiled Computing, Inc."
   index: 3
 - name: "NASA Langley Research Center, Atmospheric Science Data Center, Hampton, VA, USA"
   index: 4
   ror: 0399mhs52
 - name: "Booz Allen Hamilton, Inc., McLean, VA, USA"
   index: 5
   ror: 051rcp357
 - name: "University of Alaska Fairbanks, Fairbanks, AK, USA"
   index: 6
   ror: 01j7nq853
 - name: "Openscapes and National Center for Ecological Analysis and Synthesis (NCEAS), University of California Santa Barbara, Santa Barbara, CA, USA"
   index: 7
   ror: 02t274463
 - name: "University of New Hampshire, Durham, NH, USA"
   index: 8
   ror: 02q4hks80
 - name: "Independent Contributor"

>    index: 9

Reviewer: ```suggestion
   index: 9
 - name: "Schmidt Center for Data Science and Environment (DSE), University of California Berkeley"
   index: 10
   ror: "01an7q238"
```

are at varying stages of cloud adoption.

**Flat, functional top-level API.** All primary operations are exposed as module-level
functions (e.g., `earthaccess.login()`, `earthaccess.search_data()`,
`earthaccess.download()`), minimizing conceptual overhead for new users while
keeping the underlying object-oriented classes accessible for advanced use cases.

# Research impact statement

`earthaccess` has established itself as foundational infrastructure for NASA Earth
science data access. Concrete evidence of its impact includes:

**Peer-reviewed publications.** `earthaccess` has been used in published research,
including studies on multi-sensor drought observations in forested environments
[@andreadis2024] and tidal bore detection using SWOT satellite data [@arildsen2025].

**Community adoption.** The library is a dependency of 230 public GitHub
repositories (as of 5 March 2026), spanning data analysis workflows, Jupyter-based tutorials, and
downstream libraries. It is distributed through both PyPI and conda-forge, and has
been installed and used in cloud-hosted Jupyter environments provided by NASA and
partner organizations.

> 

Reviewer 2: Thinking of more examples to add to Community adoption, or a new section on software dependencies? Thinking of Dan Shapero's success story: working on a[ software package](https://icepack.github.io/) for ice flow modeling and replaced all his old code: 
https://github.com/icepack/icepack/commit/b3528bed78954cbd8e5b8943eafce575333f9854

three-step workflow of `login()`, `search_data()`, and `download()` -- while offering
sufficient depth for advanced users who need direct S3 access, streaming file handles,
or virtual dataset construction for large-scale analysis.

# State of the field

The scholarly contribution of `earthaccess` is the _integration_ of search,
authentication, and access into a coherent abstraction that masks the heterogeneity
of NASA Earthdata's infrastructure. No existing tool provides this end-to-end,
provider-agnostic workflow. Rather than reinventing query or filesystem libraries,
`earthaccess` composes and extends existing open-source tools and contributes the
NASA-specific domain knowledge (DAAC configurations, credential endpoints,
authentication flows, cloud-detection logic) that binds them into a usable
data access layer.

Several tools exist for accessing NASA Earth science data, each only addressing a specific
slice of the workflow:

> - **python-cmr** [@python_cmr] provides a Python wrapper around the CMR API for dataset

Author: Part of the suggestion from https://github.com/earthaccess-dev/earthaccess/pull/1249/changes#r3190761529:
```suggestion
- **python-cmr** [@python_cmr] handles the search step (1) by providing a Python wrapper around the CMR API for dataset
```

  and granule queries. `earthaccess` builds on `python-cmr`, extending it with
  DAAC-aware provider resolution, cloud-hosting filters, and rich result objects that
  encapsulate metadata. However, `python-cmr` does not handle authentication, data
  download, or cloud access -- the areas where researchers face many workflow difficulties.
  

> - **asf_search** [@asf_search] sits between `python-cmr` and `earthaccess`: it focuses on search and discovery but also handles authentication and access. It emerged around the same time as `earthaccess`, from similar access challenges. Unlike `earthaccess`, which is mission- and domain-agnostic, `asf_search` is tailored to synthetic aperture radar (SAR) data, adding domain-specific tooling and functionality.

Author: Part of the suggestion from https://github.com/earthaccess-dev/earthaccess/pull/1249/changes#r3190761529:

```suggestion
- **asf_search** [@asf_search] sits between `python-cmr` and `earthaccess`: it focuses on search and discovery but also handles authentication and access (workflow steps 1–5). It emerged around the same time as `earthaccess`, from similar access challenges. Unlike `earthaccess`, which is mission- and domain-agnostic, `asf_search` is tailored to synthetic aperture radar (SAR) data, adding domain-specific tooling and functionality.
```

> - **icepyx** [@icepyx] provides a mission-specific solution to the general problem solved by earthaccess: search, discovery, authentication, and access to ICESat-2 [@icesat2] data products. Established about a year before earthaccess, it now leverages earthaccess via a mixin to provide authentication and tokening when users perform an action that requires logging in.

Author: Part of the suggestion from https://github.com/earthaccess-dev/earthaccess/pull/1249/changes#r3190761529:
```suggestion
- **icepyx** [@icepyx] provides a mission-specific solution to the general problem solved by earthaccess: search, discovery, authentication, and access to ICESat-2 [@icesat2] data products. Established about a year before earthaccess, it handles workflow steps 1, 2, and 5 on its own and now leverages earthaccess via a mixin to provide authentication and tokening when users perform an action that requires logging in.
```

- **earthdatalogin** [@earthdatalogin_r] provides similar authentication and access

>   functionality for the R programming ecosystem. The two projects share a common motivation and

Author: Part of the suggestion from https://github.com/earthaccess-dev/earthaccess/pull/1249/changes#r3190761529:
```suggestion
- **earthdatalogin** [@earthdatalogin_r] provides similar authentication and access
  functionality (steps 3–5) for the R programming ecosystem. The two projects share a common motivation and
```

streaming data directly into analysis-ready formats using `fsspec` [@fsspec] and
constructing virtual Zarr stores from archival formats (e.g., HDF5 and NetCDF4) using
DMR++ metadata [@dmrpp], powered by VirtualiZarr [@virtualizarr] and kerchunk [@kerchunk], enabling drastic improvements in access performance.

# Statement of need

NASA's Earth science data archive is one of the largest and most diverse collections of
Earth observation data in the world, used by over ten million researchers, educators,
and decision-makers globally [@nasa_esds_data_metrics]. However, the complexity of the underlying data infrastructure
presents a significant barrier to scientific productivity. A typical data access workflow
requires a researcher to: (1) authenticate with NASA Earthdata Login; (2) discover
relevant datasets and granules through the CMR API; (3) parse metadata to obtain access
URLs; (4) manage HTTP sessions with tokens and redirect handling; (5) determine whether
data are hosted on-premises or in the Earthdata Cloud; and (6) obtain temporary AWS S3
credentials when accessing cloud-hosted data. Each step introduces opportunities for

> error, and DAAC-specific configurations further compound the challenge.

Author: To combine with the above change.
```suggestion
```

    affiliation: "11"
  - given-names: "Amy"
    surname: "Steiker"
    orcid: "https://orcid.org/0000-0002-3039-0260"
    affiliation: "1"
  - given-names: "Sherwin"
    surname: "Varghese"
    orcid: "https://orcid.org/0009-0005-7030-213X"
    affiliation: "12"
affiliations:
 - name: "National Snow and Ice Data Center (NSIDC), Cooperative Institute for Research in Environmental Sciences (CIRES), University of Colorado Boulder, Boulder, CO, USA"
   index: 1
   ror: 02s376052
 - name: "NASA Goddard Space Flight Center, Goddard Earth Sciences Data and Information Services Center (GES DISC), Greenbelt, MD, USA"
   index: 2
   ror: 0171mag52
 - name: "Coiled Computing, Inc."
   index: 3
 - name: "Ocean Ecology Lab, NASA Goddard Space Flight Center, Greenbelt, MD, USA"
   index: 4
   ror: 0171mag52
 - name: "Development Seed, Inc."
   index: 5
 - name: "Schmidt Center for Data Science and Environment (DSE), University of California Berkeley"
   index: 6
   ror: "01an7q238"
 - name: "NASA Langley Research Center, Atmospheric Science Data Center (ASDC), Hampton, VA, USA"
   index: 7
   ror: 0399mhs52
 - name: "Booz Allen Hamilton, Inc., McLean, VA, USA"
   index: 8
   ror: 051rcp357

>  - name: "University of Alaska Fairbanks, Fairbanks, AK, USA"

Reviewer 3: ```suggestion
 - name: "Alaska Satellite Facility, Geophysical Institute, University of Alaska Fairbanks, Fairbanks, AK, USA"
```

experience.

`earthaccess` provides a unified, high-level Python interface that reduces this
workflow to just a few lines of code. The library handles authentication with NASA's
Earthdata Login (EDL) service [@nasa_edl], exposes NASA's Common Metadata Repository
(CMR) [@nasa_cmr] for data discovery, and transparently manages data retrieval via
either HTTPS download or direct S3 access when running in the Amazon Web Services (AWS)
`us-west-2` region -- where data within NASA's Earthdata Cloud reside. `earthaccess` also supports
streaming data directly into analysis-ready formats using `fsspec` [@fsspec] and
constructing virtual Zarr stores from archival formats (e.g., HDF5 and NetCDF4) using
DMR++ metadata [@dmrpp], powered by VirtualiZarr [@virtualizarr] and kerchunk [@kerchunk], enabling drastic improvements in access performance.

# Statement of need

NASA's Earth science data archive is one of the largest and most diverse collections of
Earth observation data in the world, used by over ten million researchers, educators,
and decision-makers globally [@nasa_esds_data_metrics]. However, the complexity of the underlying data infrastructure

> presents a significant barrier to scientific productivity. A typical data access workflow

Reviewer 3: I'm not sure I like how we've ordered the "data access workflow". Right now we have:
1. auth
2. "discover"
3. parse metadata
4. sessions + redirects
5. is cloud?
6. S3 credentials

I think (1) and (4) should be combined and indeed that's how we discuss it on L196
https://github.com/earthaccess-dev/earthaccess/pull/1249/changes#diff-e504eb580b095a7e65428b098183a581e475f0fb316db95287eacd7d4f344424R196

Similarly, (5) and (6) are also optional and only for in-place cloud access with performance constraints or if you want to use S3 aware tools, and really, fit into (1) and (4) as well, which is also discussed this way on L196.

I also think (2) is better described as "search" and (2) + (3) is what I would call discovery. At least for me, I am always parsing metadata as part of what I'd call discovery -- typically searching broadly and then refining with sensor/bands/variable/etc, so that I end up with the actual set of granules I want to use in my workflow. I don't really see why getting the access URLs are special compared to getting any of the other metadata along the way.

We don't talk about data preparation at all, except as features of Harmony and Icepyx, which seems like a missed opportunity. 

I think I'd restructure this like:
1. Discovery
2. Auth (EDL, S3, Sessions + redirects)
3. Access
4. Data prep (includes virtual datasets and transformations)

which is similar to the Software Design section. Note, I've put auth after discovery since you generally only need it to access data, unless you're trying to discover restricted datasets... so It could go before or after discovery, I think it just flows a little better narrative-ly after, but 🤷 .

Author: I think this is a great revision you're suggesting, @jhkennedy. My immediate gut thought about ordering was that Auth goes before Discovery, but that's mostly because in code I usually run `earthaccess.login()` before anything else. Your point about it really only being needed to *access* data (restricted datasets aside) is enough for me to feel comfortable putting it where you suggest. And you're right that as written the list doesn't totally align with how we describe things later at line 196 – auth and the session/redirect handling are really one concern, and the cloud-detection and S3-credential steps are really a single conditional concern.

The one spot where I hesitate is restructuring this Statement-of-Need section to exactly match the four-layer Software Design section (Discovery / Auth / Access / Data prep). I believe the two sections serve different purposes: _Statement-of-Need_ showing how much a researcher has to deal with today, where part of the point is the number of separate steps involved, while _Software-Design_ shows the cleaner abstraction `earthaccess` provides on top of that. If both sections use the same four-stage structure, it seems to me that the problem statement gets slightly weaker as the two sections overlap more.

### Revised suggestion
So, I suppose I'd lean toward
- keeping it as a numbered list of steps (still one fewer than before [5 instead of 6]: Search / Metadata / Auth / S3 setup / Access),
- putting discovery first (before auth) as you suggested, and
- fixing the other clarification points you raised (auth + session handling as one thing; cloud-detection and S3-credential steps merged into a single conditional step).

Even though I agree "search" and "parse metadata" are together discovery, I think part of the challenge in other workflows is that they are handled separately – and so that can be part of the statement of need. So, how about something like this?

> A typical data access workflow requires a researcher to: (1) search the CMR API to discover relevant datasets and granules; (2) parse the returned metadata to extract the appropriate access URLs; (3) authenticate with NASA Earthdata Login and maintain authenticated sessions that handle tokens and cross-domain redirects; (4) for cloud-hosted data they wish to access directly while working in-region, determine that the data are cloud-hosted and obtain and periodically renew temporary AWS S3 credentials; and (5) retrieve the data, either by downloading it over HTTPS or streaming it directly from S3.

### On data prep

On _data preparation_, I'm a bit hesitant to add it as a stage here, since transformation/subsetting isn't really something `earthaccess` itself does (as opposed to Harmony/icepyx). Listing it as a step might even read as if we're highlighting a gap in what `earthaccess` covers. I think we'd be better off referring to it in passing in the State-of-the-field section, which connects to your other comments there about tying tools to workflow steps. What do you think?  Do you still want a data-prep step in this list, or should we keep it light?

**Other Thoughts?  Happy to move towards the four-stage structure you suggested earlier, if you think that's clearer than this revision.**

Reviewer 4: Great discussion! I really like where this is going. I'm going to build on @danielfromearth's suggested rewrite:
> A typical data access workflow requires a researcher to: (1) search the CMR API to discover relevant datasets and products; (2) parse the returned metadata to identify specific granules and extract the appropriate access URLs; (3) authenticate with NASA Earthdata Login and maintain authenticated sessions that handle tokens and cross-domain redirects; (4) for in-region, Earthdata cloud-hosted data, confirm that the data are cloud-hosted and obtain and periodically renew temporary AWS S3 credentials; and (5) retrieve the data by downloading it over HTTPS, streaming it directly from S3, or passing the information into a third-party package for customization services. Researchers must know enough about the technical implementation of multiple APIs to perform these steps accurately, each of which introduces opportunities for error. DAAC-specific configurations and NASA's ongoing migration to the Earthdata Cloud compound the challenge, as researchers contend with multiple access paradigms, often within a single analysis workflow.

Then we start with the "During workshops organized..." as a new paragraph.

Author: That sounds good to me!  

@jhkennedy?

Reviewer 3: That sounds good to me as well. The only thing I'd change is (4):
>  (4) for Earthdata cloud-hosted data, obtain and periodically renew temporary AWS S3 credentials

or to keep the in-region caveat, something like:
> (4) for Earthdata hosted in the same AWS region you're working in, obtain and periodically renew temporary AWS S3 credentials

because of the reasons I mentioned here:
https://github.com/earthaccess-dev/earthaccess/pull/1249#pullrequestreview-4230623467

> I have a pretty big concern around publishing this discussing the automatic cloud-detection logic. That's something we know is technically infeasible to do reliably and we've decided to rip out:
https://github.com/earthaccess-dev/earthaccess/blob/main/docs/governance/decisions/231-aws-us-west-2-checking-method.md
> 
> So I'd like to either not mention it or abstract that away in the manuscript language.

Also, within the narrative, deciding which URLs to work with happens in (2) so by (4) you've already made this decision anyways.

Author: How about a slight change of wording to keep the _data_ outside of _Earthdata_ (as the NASA program/project), and to keep the researcher/3rd-person point of view:

"for Earthdata <ins>**Cloud data**</ins> hosted in the same AWS region <ins>**as the researcher's compute environment**</ins>, obtain and periodically renew temporary AWS S3 credentials

@jhkennedy , how does that sound to you?

```suggestion
presents a significant barrier to scientific productivity. A typical data access workflow requires a researcher to: (1) search the CMR API to discover relevant datasets and products; (2) parse the returned metadata to identify specific granules and extract the appropriate access URLs; (3) authenticate with NASA Earthdata Login and maintain authenticated sessions that handle tokens and cross-domain redirects; (4) for Earthdata Cloud data hosted in the same AWS region as the researcher's compute environment, obtain and periodically renew temporary AWS S3 credentials; and (5) retrieve the data by downloading it over HTTPS, streaming it directly from S3, or passing the information into a third-party package for customization services. Researchers must know enough about the technical implementation of multiple APIs to perform these steps accurately, each of which introduces opportunities for error. DAAC-specific configurations and NASA's ongoing migration to the Earthdata Cloud compound the challenge, as researchers contend with multiple access paradigms, often within a single analysis workflow.
```

Reviewer 3: That works for me! 

nitpicky wise, I don't think users should really care or know about the difference between:
* NASA Earthdata (the project)
* NASA Earthdata Cloud (where we store the data)
* NASA Earthdata Cloud data (the actual data)

All that is just "NASA Earthdata" to me and NASA Earthdata Cloud data is a mouthful. I guess if you want to *not* use Earthdata there the alternative I would use is "Earth science data" which is what's done on Earthdata Search's landing page:
https://search.earthdata.nasa.gov/
 
 > EARTHDATA SERACH
 > Search NASA's 2.2 billion+ Earth observations
 > Use keywords and filter by time and spatial area to search NASA's **Earth science data**
 
 And is how it's termed on the main Earthdata page:
 https://www.earthdata.nasa.gov/
 
 > EARTHDATA
 > Your Gateway to NASA  Earth Observation Data
> The Earth Science Data Systems (ESDS) Program provides open access to NASA’s archive of **Earth science data**, empowering researchers and decision makers to better understand and protect our home planet.
 
 (I like Earth Science data over Earth observation data because CMR has more than observational data in it)

Author: "Earth science data" sounds good to me too.

```suggestion
presents a significant barrier to scientific productivity. A typical data access workflow requires a researcher to: (1) search the CMR API to discover relevant datasets and products; (2) parse the returned metadata to identify specific granules and extract the appropriate access URLs; (3) authenticate with NASA Earthdata Login and maintain authenticated sessions that handle tokens and cross-domain redirects; (4) for NASA Earth science data hosted in the same AWS region as the researcher's compute environment, obtain and periodically renew temporary AWS S3 credentials; and (5) retrieve the data by downloading it over HTTPS, streaming it directly from S3, or passing the information into a third-party package for customization services. Researchers must know enough about the technical implementation of multiple APIs to perform these steps accurately, each of which introduces opportunities for error. DAAC-specific configurations and NASA's ongoing migration to the Earthdata Cloud compound the challenge, as researchers contend with multiple access paradigms, often within a single analysis workflow.
```

requires a researcher to: (1) authenticate with NASA Earthdata Login; (2) discover

> relevant datasets and granules through the CMR API; (3) parse metadata to obtain download

Reviewer 3: ```suggestion
relevant datasets and granules through the CMR API; (3) parse metadata to obtain access
```

This is true for downloading, in-place HTTP access, or S3 "direct" access"

URLs; (4) manage HTTP sessions with tokens and redirect handling; (5) determine whether
data are hosted on-premises or in the Earthdata Cloud; and (6) obtain temporary AWS S3

> credentials when accessing cloud-hosted data. Each step introduces opportunities for

Reviewer 3: You only need to do (5) if you're doing (6)...

Reviewer 4: marking resolved since we address it here: https://github.com/earthaccess-dev/earthaccess/pull/1249/changes#r3190909635

error, and DAAC-specific configurations further compound the challenge.

NASA's ongoing migration to the Earthdata Cloud adds further complexity, as researchers

> must now contend with two possible access paradigms, traditional HTTPS downloads and S3-based

Reviewer 3: I think this sentence should be moved up into the previous paragraph before (5) and (6), or part of a stand alone paragraph with (5) and (6).

Reviewer 4: would be resolved by https://github.com/earthaccess-dev/earthaccess/pull/1249/changes#r3538571932

access. These both may even occur within a single analysis workflow. During workshops organized by NASA
Openscapes [@nasa_openscapes; @lowndes2019], the need for simpler tools became evident.

> `earthaccess` was created to address this gap: it provides uniform access to NASA

Reviewer 3: ```suggestion
`earthaccess` is a community project that was created to address this gap: it provides uniform access to NASA
```

We don't mention community at all until the end of the Software Design section, and we don't talk about the community aspect of developing this library at all, which I think is pretty integral to it's success and would be nice to represent somewhere in the introduction.

three-step workflow of `login()`, `search_data()`, and `download()` -- while offering
sufficient depth for advanced users who need direct S3 access, streaming file handles,
or virtual dataset construction for large-scale analysis.

# State of the field

The scholarly contribution of `earthaccess` is the _integration_ of search,
authentication, and access into a coherent abstraction that masks the heterogeneity
of NASA Earthdata's infrastructure. No existing tool provides this end-to-end,
provider-agnostic workflow. Rather than reinventing query or filesystem libraries,
`earthaccess` composes and extends existing open-source tools and contributes the
NASA-specific domain knowledge (DAAC configurations, credential endpoints,
authentication flows, cloud-detection logic) that binds them into a usable
data access layer.

Several tools exist for accessing NASA Earth science data, each only addressing a specific

> slice of the workflow:

Reviewer 3: The following list reads a little funny to me. We provide the 4-6 steps of a workflow before here:
1. auth
2. discovery
3. parse metadata
4. sessions and tokens
5. is cloud?
6. S3 credentials

But we talk about specific tools and don't directly connect them to these (6) steps. 

I'd prefer to I think group tools under the things they do, or at least call it out directly like:

- **python-cmr** [@python_cmr] handles step (2) by providing a Python wrapper...

Author: @jhkennedy, I like the idea of connecting the descriptions back to the outlined steps in the "statement of need" section! 

(Though I wouldn't want a reader to **_need to_** flip back and forth between this and the "statement of need" sections to understand the tool entries. So, even though it means a tiny bit of redundancy, I'm thinking we keep descriptions **and** include a parenthetical to the step number.)

Author: ```suggestion
Several tools exist for accessing NASA Earth science data, each addressing only part of
the workflow, or addressing the full workflow for only a subset of NASA's data holdings:
```

Author: @jhkennedy, how does this and the suggested changes for the following tool entries sound to you?

- **python-cmr** [@python_cmr] provides a Python wrapper around the CMR API for dataset
  and granule queries. `earthaccess` builds on `python-cmr`, extending it with
  DAAC-aware provider resolution, cloud-hosting filters, and rich result objects that
  encapsulate metadata. However, `python-cmr` does not handle authentication, data

>   download, or cloud access -- the areas where researchers face many workflow difficulties.

Reviewer 3: We should also call out `asf_search` -- it's in between python_CMR and Earthaccess, focused on search and discovery but handles auth/etc. It is however, primarily focused on SAR data so has domain-specific tools/functionality added to it.

It was started 2 months before Earthaccess and came out of the same need/problems but with a different focus

Author: Sounds good to me!  @jhkennedy, here's a rough pass at an `asf_search` entry directly from the things you noted. Please rewrite however and as much as you see fit.

> - **asf_search** [@asf_search] sits between `python-cmr` and `earthaccess`: it focuses on search and discovery but also handles authentication and access. It emerged around the same time as `earthaccess`, from similar access challenges. Unlike `earthaccess`, which is mission- and domain-agnostic, `asf_search` is tailored to synthetic aperture radar (SAR) data, adding domain-specific tooling and functionality.

And here's a @software BibTeX entry to reference, matching the style of the other tool citations. It points to the Python package rather than the search.asf.alaska.edu web app, since the other State‑of‑the‑field paragraphs compare Python libraries...
```
@software{asf_search,
  title = {asf\_search: {Python} wrapper for the {NASA} {Alaska Satellite Facility} {SearchAPI}},
  author = {{Alaska Satellite Facility}},
  year = {2026},
  url = {https://github.com/asfadmin/Discovery-asf_search}
}
```

Reviewer 3: That sound good to me!

Author: ```suggestion
- **python-cmr** [@python_cmr] provides a Python wrapper around the CMR API for dataset
  and granule queries. `earthaccess` builds on `python-cmr`, extending it with
  DAAC-aware provider resolution, cloud-hosting filters, and rich result objects that
  encapsulate metadata. However, `python-cmr` does not handle authentication, data
  download, or cloud access -- the areas where researchers face many workflow difficulties.
  
- **asf_search** [@asf_search] sits between `python-cmr` and `earthaccess`: it focuses on search and discovery but also handles authentication and access. It emerged around the same time as `earthaccess`, from similar access challenges. Unlike `earthaccess`, which is mission- and domain-agnostic, `asf_search` is tailored to synthetic aperture radar (SAR) data, adding domain-specific tooling and functionality.
```

- **harmony-py** [@harmony_py] is NASA Earthdata's client for the Harmony data transformation
  service, which provides, for example, server-side subsetting, reformatting, and reprojection services.
  It addresses a complementary use case; `earthaccess` focuses on direct data access

>   and client-side analysis.

Reviewer 3: In the context of this paper, I'm not sure why we're including this here... We don't talk about any parts of the workflow after you access the data in our six steps, and this is a follow on service. 

If we want to include it, we should really add a seventh "data preparation" step to the workflow which would cover harmony and icepyx's subsetting and variable selection, as well as other tools we haven't mentioned like HyP3.

Reviewer 4: Good point. We could remove harmony-py, and reframe the icepyx one similar to asf_search:
> icepyx [@icepyx] provides a mission-specific solution to the general problem solved by `earthaccess`: search, discovery, authentication, and access to ICESat-2 [@icesat2] data products. Established about a year before `earthaccess`, it now leverages `earthaccess` via a mixin to provide authentication and tokening when users perform an action that requires logging in.

I will work on the icesat2 ref if we go this route.

Author: Removing `harmony-py` sounds good to me.

@JessicaS11, that would be great if you could tackle the icesat2 ref.

```suggestion
```

Reviewer 4: > that would be great if you could tackle the icesat2 ref.

Added as a suggested edit to paper.bib.

- **icepyx** [@icepyx] provides specialized tools for ICESat-2 data, including
  subsetting and variable selection. It is mission-specific by design, whereas
  `earthaccess` is mission-agnostic and supports all EOSDIS data holdings.

- **earthdatalogin** [@earthdatalogin_r] provides similar authentication and access
  functionality for the R programming ecosystem. The two projects share a common motivation and

>   serve as complementary tools for their respective language communities.

Reviewer 3: 🤔 are there other R/Julia things we should call out?

Author: Happy to add others - anyone know?

- **Direct use of `fsspec`/`s3fs`** [@fsspec; @s3fs]: Advanced users can compose their
  own access workflows using these general-purpose filesystem libraries. However, this
  requires extensive knowledge of NASA's authentication flow, DAAC-specific credential

>   endpoints, and the mapping between on-premises and cloud-hosted data links.

Reviewer 4: ```suggestion
  endpoints, and the mapping between download versus cloud-streaming data access links.
```

Author: Hmm, that sounds okaaay to me, though I'm not sure which is more accurate. Other opinions?

Reviewer 5: I like Jessica's suggestion. on-premises is a vague term for non technical people.

# Software design

`earthaccess` is organized into four core layers, each encapsulating a distinct
component of the data access workflow:

1. **Authentication**: Manages the full lifecycle of NASA Earthdata Login credentials,
   supporting environment variables, `.netrc` files, and interactive prompts. Once
   authenticated, the library creates HTTP sessions that correctly handle NASA's
   cross-domain redirects and retrieves and renews temporary AWS S3 credentials for in-region
   cloud access.

2. **Search**: Extends `python-cmr` with DAAC-aware provider resolution, cloud-hosting
   detection, and deep-paging support. Query results are wrapped in rich objects that
   preserve the full metadata response while exposing convenience methods for data
   links, spatial footprints, and formatted citations.

3. **Access**: Attempts to detect at runtime whether the process is running within AWS `us-west-2`
   and automatically selects the optimal access path -- direct S3 reads for in-region

>    access or HTTPS downloads otherwise. Users can manually specify an access path if needed.  Files can be opened as `fsspec`-compatible

Reviewer 5: I like the concise way of presenting this, maybe we can add that being format-agnostic and python file-like object compatible makes the library interoperable with the rest of the scientific python ecosystem (aka Pydata/Pangeo)

>    access or HTTPS downloads otherwise. Users can manually specify an access path if needed.  Files can be opened as `fsspec`-compatible

Reviewer 3: ```suggestion
   access or HTTPS access otherwise. Users can manually specify an access path if needed.  Files can be opened as `fsspec`-compatible
```

You can download or stream via HTTPS

   file-like objects for streaming into libraries such as xarray [@xarray], or
   downloaded to disk with parallel, fault-tolerant transfers.

4. **Virtual datasets**: Leverages NASA's DMR++ sidecar metadata files [@dmrpp] to
   construct virtual Zarr stores via VirtualiZarr [@virtualizarr] or kerchunk
   [@kerchunk], enabling lazy, chunk-level access to archival HDF5/NetCDF4 data without downloading or reformatting files. For example, a researcher can extract a single variable across thousands of files by reading only the relevant byte ranges from NASA's cloud storage, with minimal local resource usage. These features are available as optional dependencies to keep the core library lightweight.

Several deliberate design decisions shape the library:

**Build on, don't replace, existing libraries.** `earthaccess` composes existing
open-source tools -- `python-cmr` for search, `fsspec` and `s3fs` for file I/O,
VirtualiZarr and kerchunk for virtual datasets -- rather than reimplementing their
functionality. The library's unique contribution is the NASA-specific integration

> layer that binds these tools together.

Reviewer 5: This is the awesomeness, integrating and simplifying the steps a scientist usually do when working with NASA data. Maybe adding an example of time to science reduction both in lines of code and speed through performance optimizations via fsspec and virtualizarr. Tempo or ICESat-2 can be used for this, before N minutes, now N seconds. Before 10 lines of code, now 1.

Author: Hmm, @betolink, I'm not really sure what would be appropriate for **_pre-earthaccess_** code if we want that compared with **_earthaccess-enabled_** code... would it be something like [this TEMPO S3 Access notebook](https://github.com/nasa/ASDC_Data_and_User_Services/blob/main/TEMPO/additional_drafts/S3-Access-TEMPO-L3_v4.ipynb), which uses `boto3`, `requests`, `s3fs`, etc?

Reviewer 5: Sorry for the late reply, yes that would be a good comparison, the boto3, requests, CMR response parsing, s3s credentials etc. @danielfromearth

Author: @betolink, I ran some benchmark tests, taking the median of three runs comparing `s3.download_file(...)` to `xr.open_mfdataset(earthaccess.open(results), ...)`, on the Openscapes 2i2c Jupyterhub. Here is a description of the results that we could add to the paper:

```suggestion
layer that binds these tools together. For example, retrieving NASA's TEMPO Level-3
ozone product [@tempo_o3tot_l3] directly from the Earthdata Cloud with `boto3`, `requests`, and `s3fs`
requires about two dozen lines of code (excluding imports) to set up Earthdata Login
authentication, request and periodically refresh temporary S3 credentials, and
serially transfer each granule. The equivalent `earthaccess` workflow is about four
lines (`login()`, `search_data()`, `open()`, and opening the result with `xarray`)
and additionally handles CMR-based discovery. Because
that integration also composes `fsspec` for byte-range access, streaming a single
variable across 28 granules (6.7 GB) and computing a summary statistic completed in
roughly 45 seconds end-to-end, compared with roughly 78 seconds to download the full
files serially beforehand (medians of three runs on the NASA-Openscapes `us-west-2`
JupyterHub). Constructing or reading virtual data stores over the same files with VirtualiZarr [@virtualizarr], which enables reading only the requested chunks, can reduce access time further. The same pattern applies to other
granule-heavy products, such as ICESat-2 ATL06 land-ice height data [@icesat2], and is
most pronounced where an analysis touches only a small portion of each file.
```
I would add a corresponding citation for the data too, in the `paper.bib`.

What do you think of this?

that `earthaccess` depends on.

**Location-transparent access.** The same user code works whether the computation runs
in the cloud or on a local workstation. The library automatically selects the optimal
access path without requiring code changes, reflecting the reality that researchers
are at varying stages of cloud adoption.

**Flat, functional top-level API.** All primary operations are exposed as module-level
functions (e.g., `earthaccess.login()`, `earthaccess.search_data()`,
`earthaccess.download()`), minimizing conceptual overhead for new users while
keeping the underlying object-oriented classes accessible for advanced use cases.

# Research impact statement

`earthaccess` has established itself as foundational infrastructure for NASA Earth
science data access. Concrete evidence of its impact includes:

**Peer-reviewed publications.** `earthaccess` has been used in published research,
including studies on multi-sensor drought observations in forested environments
[@andreadis2024] and tidal bore detection using SWOT satellite data [@arildsen2025].

**Community adoption.** The library is a dependency of 230 public GitHub

> repositories (as of 5 March 2026), spanning data analysis workflows, Jupyter-based tutorials, and

Reviewer 5: Let's mention machine learning projects here, some of the projects using earthaccess do AI or ML workflows even at production scale.

Reviewer 5: Each of these projects didn't have to reinvent the wheel to access NASA Earth data.

Reviewer 5: Oh nevermind, is mentioned below

NOAA, University of New Hampshire, University of Maryland), private industry (Coiled, Development Seed),
and independent open-source contributors. This breadth reflects both the library's
relevance across domains and the health of its contributor community.

**Integration with the NASA ecosystem.** `earthaccess` is featured in the official NASA
Earthdata tools catalog (<https://www.earthdata.nasa.gov/data/tools/earthaccess>) and in other NASA Earthdata tutorials. It has been presented at multiple large professional meetings -- including several American Geophysical Union (AGU) Annual Meetings and Earth System Information Partnership (ESIP) meetings -- and was the subject of
a NASA ESDS Tech Spotlight presentation (see <https://earthaccess.readthedocs.io> homepage for links to recent slides and recordings). The documentation includes executable Jupyter
notebooks demonstrating workflows with ICESat-2, EMIT, TEMPO, SMAP, and other missions,
providing reproducible entry points for researchers.

# AI usage disclosure

> No generative AI tools were used in the development of the `earthaccess` software; all architectural and design decisions were made exclusively by the authors and contributors.

Reviewer 3: Hmm, is this true anymore? @betolink have you been using Claude for the virtulizarr work?

I wonder if we need to adopt an AI policy and say something like "...developers may use AI tools but are responsible for their contributions...".

Author: I don't see any problem with modifying this to include development work that leveraged AI.  Regarding an AI policy: this question might give another nudge to the AI policy discussion of https://github.com/earthaccess-dev/earthaccess/discussions/1294.

Author: Now this AI policy is being worked on in https://github.com/earthaccess-dev/earthaccess/pull/1419.

**Integration with the NASA ecosystem.** `earthaccess` is featured in official NASA
Earthdata tutorials including <https://www.earthdata.nasa.gov/data/tools/earthaccess>, has been presented at AGU Fall Meetings, and was the subject of
a NASA ESDS Tech Spotlight presentation. The documentation includes executable Jupyter
notebooks demonstrating workflows with ICESat-2, EMIT, TEMPO, SMAP, and other missions,
providing reproducible entry points for researchers.

# AI usage disclosure

No generative AI tools were used in the development of the earthaccess software; all architectural and design decisions were made exclusively by the authors and contributors.

Preparation of this manuscript was assisted by Claude [Sonnet 4.6 and Opus 4.6] (Anthropic), which was applied to the full manuscript using the repository source code, documentation, commit history, and project metadata as context. Assistance was used for manuscript drafting and editorial revision. All content was reviewed, revised, and verified for accuracy by the authors, who bear full responsibility for the submitted work.

# Acknowledgements

The development of `earthaccess` was supported by NASA's Earth Science Data Systems

> (ESDS) program through the Openscapes project (NASA award **______**, PIs Julia

Reviewer 2: thanks for the nudge!

  DAAC-aware provider resolution, cloud-hosting filters, and rich result objects that
  encapsulate metadata. However, `python-cmr` does not handle authentication, data
  download, or cloud access -- the areas where researchers face many workflow difficulties.
  
- **asf_search** [@asf_search] sits between `python-cmr` and `earthaccess`: it focuses on search and discovery but also handles authentication and access. It emerged around the same time as `earthaccess`, from similar access challenges. Unlike `earthaccess`, which is mission- and domain-agnostic, `asf_search` is tailored to synthetic aperture radar (SAR) data, adding domain-specific tooling and functionality.

- **harmony-py** [@harmony_py] is NASA Earthdata's client for the Harmony data transformation
  service, which provides, for example, server-side subsetting, reformatting, and reprojection services.
  It addresses a complementary use case; `earthaccess` focuses on direct data access
  and client-side analysis.

- **icepyx** [@icepyx] provides specialized tools for ICESat-2 data, including
  subsetting and variable selection. It is mission-specific by design, whereas

>   `earthaccess` is mission-agnostic and supports all EOSDIS data holdings.

Author: ```suggestion
- **icepyx** [@icepyx] provides a mission-specific solution to the general problem solved by earthaccess: search, discovery, authentication, and access to ICESat-2 [@icesat2] data products. Established about a year before earthaccess, it now leverages earthaccess via a mixin to provide authentication and tokening when users perform an action that requires logging in.
```

- **Direct use of `fsspec`/`s3fs`** [@fsspec; @s3fs]: Advanced users can compose their
  own access workflows using these general-purpose filesystem libraries. However, this
  requires extensive knowledge of NASA's authentication flow, DAAC-specific credential
  endpoints, and the mapping between on-premises and cloud-hosted data links.

# Software design

`earthaccess` is organized into four core layers, each encapsulating a distinct
component of the data access workflow:

1. **Authentication**: Manages the full lifecycle of NASA Earthdata Login credentials,
   supporting environment variables, `.netrc` files, and interactive prompts. Once
   authenticated, the library creates HTTP sessions that correctly handle NASA's
   cross-domain redirects and retrieves and renews temporary AWS S3 credentials for in-region
   cloud access.

2. **Search**: Extends `python-cmr` with DAAC-aware provider resolution, cloud-hosting
   detection, and deep-paging support. Query results are wrapped in rich objects that
   preserve the full metadata response while exposing convenience methods for data
   links, spatial footprints, and formatted citations.

> 3. **Access**: Detects at runtime whether the process is running within AWS `us-west-2`

Reviewer 6: ```suggestion
3. **Access**: Attempts to detect at runtime whether the process is running within AWS `us-west-2`
```

   and automatically selects the optimal access path -- direct S3 reads for in-region

>    access or HTTPS downloads otherwise. Files can be opened as `fsspec`-compatible

Reviewer 6: ```suggestion
   access or HTTPS downloads otherwise. Users can manually specify an access path if needed.  Files can be opened as `fsspec`-compatible
```

hand-written NSIDC DAAC data-fetching routines with `earthaccess` calls, eliminating
hard-coded URLs and custom authentication logic. `earthaccess` has replaced tens of lines of code across countless NASA data access tutorials, increasing user accessibility and reducing the amount of "getting started" overhead.

**Multi-institutional development.** Contributors span NASA's Distributed Active Archive Centers (DAACs) — including ASDC, ASF, GES DISC, LP DAAC, NSIDC DAAC, OB.DAAC, ORNL DAAC, and PO.DAAC — as well as other federal and academic institutions (USGS,
University of New Hampshire), private industry (Coiled, Development Seed),
and independent open-source contributors. This breadth reflects both the library's
relevance across domains and the health of its contributor community.

**Integration with the NASA ecosystem.** `earthaccess` is featured in the official NASA
Earthdata tutorials, including <https://www.earthdata.nasa.gov/data/tools/earthaccess>, has been presented at multiple large professional meetings (including several American Geophysical Union Annual and Earth System Information Partnership (ESIP) meetings), and was the subject of

> a NASA ESDS Tech Spotlight presentation. The documentation includes executable Jupyter

Reviewer 2: ```suggestion
a NASA ESDS Tech Spotlight presentation (see <https://earthaccess.readthedocs.io> homepage for links to recent slides and recordings). The documentation includes executable Jupyter
```

notebooks demonstrating workflows with ICESat-2, EMIT, TEMPO, SMAP, and other missions,
providing reproducible entry points for researchers.

# AI usage disclosure

No generative AI tools were used in the development of the `earthaccess` software; all architectural and design decisions were made exclusively by the authors and contributors.

Preparation of this manuscript was assisted by Claude [Sonnet 4.6 and Opus 4.6] (Anthropic), which was applied to the full manuscript using the repository source code, documentation, commit history, and project metadata as context. Assistance was used for manuscript drafting and editorial revision. All content was reviewed, revised, and verified for accuracy by the authors, who bear full responsibility for the submitted work.

# Acknowledgements

The development of `earthaccess` was supported by NASA's Earth Science Data Systems

> (ESDS) program through the Openscapes project (NASA award **______**, PIs Julia

Reviewer 2: ```suggestion
(ESDS) program through the NASA Openscapes project (NASA award #20-TWSC20-2-0003 “Openscapes: Enabling a new era of science on the cloud, in response to the 2020 NASA ROSES Element E2”,
```

---

# Summary

`earthaccess` is an open-source Python library that simplifies the discovery, authentication,
and access of NASA Earth science data. NASA's Earth Observing System Data and Information System
(EOSDIS) distributes over 100 petabytes of data across 12 Distributed Active Archive Centers
(DAACs) [@nasa_earthdata], encompassing satellite imagery, climate records, atmospheric
measurements, and other geospatial datasets critical to Earth science research. Accessing these
data programmatically has historically required researchers to navigate multiple authentication
systems, understand provider-specific APIs and protocols, and write substantial boilerplate code
-- challenges that particularly affect researchers without deep software engineering
experience.

`earthaccess` provides a unified, high-level Python interface that reduces this
workflow to just a few lines of code. The library handles authentication with NASA's
Earthdata Login (EDL) service [@nasa_edl], exposes NASA's Common Metadata Repository
(CMR) [@nasa_cmr] for data discovery, and transparently manages data retrieval via
either HTTPS download or direct S3 access when running in the Amazon Web Services (AWS)

> `us-west-2` region -- where NASA's cloud-hosted data resides. `earthaccess` also supports

Author: yep, I think this is now covered nicely below in the "Software Design" access section.

relevant datasets and granules through the CMR API; (3) parse metadata to obtain download
URLs; (4) manage HTTP sessions with tokens and redirect handling; (5) determine whether
data are hosted on-premises or in the Earthdata Cloud; and (6) obtain temporary AWS S3
credentials when accessing cloud-hosted data. Each step introduces opportunities for
error, and DAAC-specific configurations further compound the challenge.

NASA's ongoing migration to the Earthdata Cloud adds further complexity, as researchers
must now contend with two possible access paradigms (traditional HTTPS downloads and S3-based
access), and sometimes even within a single analysis workflow. During workshops organized by NASA
Openscapes [@nasa_openscapes; @lowndes2019], the need for simpler tools became evident.
`earthaccess` was created to address this gap: it provides uniform access to NASA
Earthdata regardless of data storage location, enabling researchers to focus on science
rather than data engineering.

The target audience includes Earth scientists, remote sensing researchers, climate modelers,
hydrologists, ecologists, and any researcher, application developer, or educator who needs

> to work with NASA Earth science data. The library is designed to be approachable for those new to Python -- with a

Author: @asteiker, @JessicaS11, 
I don't know, this list is already quite long. Maybe we should go the other direction – consolidate it to be shorter, but not as short as the original?  How about this:

```suggestion
The target audience spans Earth system science — from atmospheric scientists and oceanographers to ecologists and land surface modelers — as well as operational communities such as weather forecasters and disaster response practitioners. It also serves remote sensing researchers, application developers, educators, and decision-makers who work with NASA Earth science data. The library is designed to be approachable for those new to Python -- with a
```

> to work with NASA Earth science data. The library is designed to be approachable for those new to Python -- with a

Reviewer 6: I like this update!

  own access workflows using these general-purpose filesystem libraries. However, this
  requires extensive knowledge of NASA's authentication flow, DAAC-specific credential
  endpoints, and the mapping between on-premises and cloud-hosted data links.

# Software design

`earthaccess` is organized into four core layers, each encapsulating a distinct
concern of the data access workflow:

1. **Authentication**: Manages the full lifecycle of NASA Earthdata Login credentials,
   supporting environment variables, `.netrc` files, and interactive prompts. Once
   authenticated, the library creates HTTP sessions that correctly handle NASA's
   cross-domain redirects and retrieves temporary AWS S3 credentials for in-region
   cloud access.

2. **Search**: Extends `python-cmr` with DAAC-aware provider resolution, cloud-hosting
   detection, and deep-paging support. Query results are wrapped in rich objects that
   preserve the full metadata response while exposing convenience methods for data
   links, spatial footprints, and formatted citations.

3. **Access**: Detects at runtime whether the process is running within AWS `us-west-2`

>    and automatically selects the optimal access path -- direct S3 reads for in-region

Reviewer 7: We still attempt to detect whether or not the code is running within the us-west-2 region, but it is still (and always will be) unreliable.

Perhaps this wording should be enhanced to say "attempts to detect," but can be user-specified in the face of failure to auto-detect.

they could benefit the wider community. This upstream-first discipline avoids
scope creep, reduces maintenance burden, and strengthens the broader ecosystem
that `earthaccess` depends on.

**Location-transparent access.** The same user code works whether the computation runs
in the cloud or on a local workstation. The library automatically selects the optimal
access path without requiring code changes, reflecting the reality that researchers
are at varying stages of cloud adoption.

**Flat, functional top-level API.** All primary operations are exposed as module-level
functions (e.g., `earthaccess.login()`, `earthaccess.search_data()`,
`earthaccess.download()`), minimizing conceptual overhead for new users while
keeping the underlying object-oriented classes accessible for advanced use cases.

# Research impact statement

`earthaccess` has established itself as foundational infrastructure for NASA Earth
science data access. Concrete evidence of its impact includes:

**Peer-reviewed publications.** `earthaccess` has been used in published research,
including studies on multi-sensor drought observations in forested environments

> [@andreadis2024] and tidal bore detection using SWOT satellite data [@arildsen2025].

Reviewer 6: Cool, thanks!

> [@andreadis2024] and tidal bore detection using SWOT satellite data [@arildsen2025].

Reviewer 5: Suggestion: maybe we can use the papers linked to the repositories that have earthaccess as a dependency. @asteiker @danielfromearth

> [@andreadis2024] and tidal bore detection using SWOT satellite data [@arildsen2025].

Author: Okay @asteiker, @betolink, I tried finding some more papers using the dependents from GitHub's 'insights' tab by creating (collaboratively with Claude) a script to do the following: For each dependent **_repo_**,
1. check whether the repo contains a `CITATION.cff` file, and
2. query the OpenAlex scholarly database for works whose full text or references mention the repo's GitHub URL , with a second pass to filter false positives from generic repo names. 

Here are ones I manually checked afterwards to make sure (1) each article references the repo and (2) each repo indeed includes earthaccess. _So, note that these papers don't necessarily cite earthaccess by name; rather, they describe or use software that depends on earthaccess_:

- Scheick et al., "Software to Enable Ocean Discoveries: A Case Study With ICESat-2 and Argo", *Geoscience Data Journal* (2025), [`icesat2py/icepyx`](https://github.com/icesat2py/icepyx).
- Wu, "Leafmap: A Python package for interactive mapping and geospatial analysis...", *JOSS* (2021), [`opengeos/leafmap`](https://github.com/opengeos/leafmap).
- Joughin et al., "Ocean-induced melt volume directly paces ice loss from Pine Island Glacier", *Science Advances* (2021), [`icepack/icepack`](https://github.com/icepack/icepack).
- Mahood et al., "Country-level fire perimeter datasets (2001–2021)", *Scientific Data* (2022), [`earthlab/firedpy`](https://github.com/earthlab/firedpy).
- Durand et al., "A Framework for Estimating Global River Discharge From SWOT", *Water Resources Research* (2023), [`SWOT-Confluence/sic4dvar`](https://github.com/SWOT-Confluence/sic4dvar).
- Tankersley, "PolarToolkit: Python Tools for Convenient, Reproducible, and Open Polar Science", *JOSS* (2024), [`mdtanker/polartoolkit`](https://github.com/mdtanker/polartoolkit).
- Xiahou et al., "Hydrographic variability in the central Ross Ice Shelf cavity and the implications for ocean circulation", *JGR Oceans* (2026), [`mdtanker/polartoolkit`](https://github.com/mdtanker/polartoolkit).
- Widlansky & Komar, "Building an Intelligent Data Exploring Assistant for Geoscientists", *JGR ML & Computation* (2025), [`uhsealevelcenter/IDEA`](https://github.com/uhsealevelcenter/IDEA).
- Gardner et al., "ITS_LIVE global glacier velocity data in near-real time", *The Cryosphere* (2025), [`ASFHyP3/itslive-metadata`](https://github.com/ASFHyP3/itslive-metadata).
- Yusuf et al., "InstaGeo: Compute-Efficient Geospatial ML from Data to Deployment", *arXiv* (2025), [`instadeepai/InstaGeo-E2E-Geospatial-ML`](https://github.com/instadeepai/InstaGeo-E2E-Geospatial-ML).

Should we include some of these here, would either of you want to suggest how to work these in without it being too verbose or simply writing out a list?

> [@andreadis2024] and tidal bore detection using SWOT satellite data [@arildsen2025].

Reviewer 6: @danielfromearth This is so great to see! For some reason I can't suggest a change from this line, but How about this addition: 

"Several other research studies and geospatial technology publications utilize libraries dependent on `earthaccess`, including... " (And we should just pick two or three citations from this list)

I'm also going to reference this in #1216

> [@andreadis2024] and tidal bore detection using SWOT satellite data [@arildsen2025].

Author: Thanks @asteiker, that sounds good to me – here it is with three picked out, do they look good to you?

```suggestion
[@andreadis2024] and tidal bore detection using SWOT satellite data [@arildsen2025]. Several other research studies and geospatial technology publications utilize libraries dependent on `earthaccess`, including [@Scheick2025], [@Xiahou2026], [@Widlansky2025].
```

Reminder to ourselves: Will need to add these to the .bib file too.

**Community adoption.** The library is a dependency of 230 public GitHub
repositories (as of 5 March 2026), spanning data analysis workflows, Jupyter-based tutorials, and
downstream libraries. It is distributed through both PyPI and conda-forge, and has
been installed and used in cloud-hosted Jupyter environments provided by NASA and
partner organizations. As one example of downstream adoption, icepack -- a finite
element library for ice sheet and glacier modeling [@shapero2021] -- replaced its
hand-written NSIDC data-fetching routines with `earthaccess` calls, eliminating
hard-coded URLs and custom authentication logic.

**Multi-institutional development.** Contributors span NASA's Distributed Active Archive Centers (DAACs) — including ASDC, ASF, GES DISC, LP DAAC, NSIDC, OB.DAAC, ORNL DAAC, and PO.DAAC — as well as other federal and academic institutions (USGS,

> University of New Hampshire), private industry (Coiled, Development Seed),

Reviewer 4: I like the general descriptor and then examples in parenthesis format. How about:

```suggestion
NOAA, University of New Hampshire, University of Maryland), private industry (Coiled, Development Seed),
```

and independent open-source contributors. This breadth reflects both the library's
relevance across domains and the health of its contributor community.

**Integration with the NASA ecosystem.** `earthaccess` is featured in official NASA

> Earthdata tutorials, including <https://www.earthdata.nasa.gov/data/tools/earthaccess>, has been presented at multiple large professional meetings (including several American Geophysical Union Annual and Earth System Information Partnership (ESIP) meetings), and was the subject of

Reviewer 6: I like it!
